component/core/src/main/java/org/exoplatform/social/core/metadata/attachment/AttachmentServiceImpl.java

```diff
@@ -186,6 +192,64 @@ public ObjectAttachmentList getAttachments(String objectType,
     return new ObjectAttachmentList(attachments, objectType, objectId);
   }
 
+  @Override
+  public void updateAttachments(String objectType,
+                                String objectId,
+                                List<String> fileIds,
+                                Identity userAclIdentity) throws ObjectNotFoundException, IllegalAccessException {
+
+    org.exoplatform.social.core.identity.model.Identity userIdentity =
+                                                                     identityManager.getOrCreateUserIdentity(userAclIdentity.getUserId());
+
+    if (userIdentity == null || userIdentity.isDeleted() || !userIdentity.isEnable()) {
+      throw new IllegalStateException("User with id " + userAclIdentity.getUserId() + " isn't valid");
+    }
+    if (StringUtils.isBlank(objectType)) {
+      throw new IllegalArgumentException("Object type is mandatory");
+    }
+    if (StringUtils.isBlank(objectId)) {
+      throw new IllegalArgumentException("Object identifier is mandatory");
+    }
+    for (String fileId : fileIds) {
+      if (StringUtils.isBlank(fileId)) {
+        throw new IllegalArgumentException("File id is mandatory");
+      }
+      if (!hasEditPermission(userAclIdentity, objectType, objectId)) {
+        throw new IllegalAccessException("User " + userAclIdentity.getUserId()
+            + " doesn't have enough permissions to delete file " + fileId + " from on object "
+            + objectType + "/" + objectId);
+      }
+    }
```

Comment:
No need to make the permission check in the loop, for each file

```suggestion
      }
      if (!hasEditPermission(userAclIdentity, objectType, objectId)) {
        throw new IllegalAccessException("User " + userAclIdentity.getUserId()
            + " doesn't have enough permissions to update files of object "
            + objectType + "/" + objectId);
      }
```